configs/main.yaml

```diff
@@ -7,6 +7,8 @@ train:
   label_smoothing: 0.1
 
 dataset_name: 'roszcz/maestro-v1-sustain'
+predict: velocity
+tgt_bins: 200
```

Comment:
not sure as to the "predict" name for this hyperparameter,
I also added tgt_bins to config to choose target values quantization bins number.

Reply:
Maybe `target: velocity` would be better. `tgt_bins` also need a revamp - I think we should be explicit and call it `dstart_bins`. It could be set in the `dataset.quantization` section, and if you don't want to use the same MidiQuantizer approach, I think this should be a separate config

Reply:
I could set dstart_bins in dataset.quantization if I was preparing a cached dataset with tgt_dstart_bins field. If I keep quantizng dstart during tokenization I would be caching the same object with different name (hashed config). We also do not need dstart_bins in datasets used for velocity prediction. BTW total quantization time in DstartEncoder is 2.75 seconds per epoch on whole maestro-v1-sustain dataset so I think this approach acceptable, what do you think?

Reply:
How did you measure this encoder time?

Speed is one concern, so if it's fast then that's fine. My other concern is the code entanglement - I feel like we should rip the `dstart` and `velocity` predictions apart as much as possible 🤔 

Please take a look at https://github.com/Nospoko/autoencoders-demo and let me know how do you feel about doing a similar pipeline-based design in this project

Reply:
I measured it by crating total_time atribute in DstartEncoder and used it to count time elapsed during quantization step inside tokenize function.